### PR TITLE
fix join_group trigger for private groups

### PR DESCRIPTION
**Pull request by cherbst**
also trigger join_group if user accepts an invitation
or a membership request gets accepted

### CODE REVIEW

**Comment by tw2113** [comment]
This is in conjunction with https://github.com/opencredit/BadgeOS-Community-Add-on/issues/19 correct?

**Comment by cherbst** [comment]
Correct, I did apply the patch to fix #19 before, so lines 144 to 146 come from this patch. As a result,   my fix should already work with the join_specific_group trigger.

File: includes/rules-engine.php [hunk 1]
```diff
@@ -41,6 +41,9 @@ function badgeos_bp_trigger_event( $args = '' ) {
 	if ( empty( $user_ID ) && 'bp_core_activated_user' == current_filter() )
 		$user_ID = absint( $args );
 
+	if ( 'groups_join_specific_group' == current_filter() )
+		$user_ID = absint( $args[1] );
+
```
Comment (cherbst) [inline includes/rules-engine.php]: These lines are only really needed for the `groups_membership_accepted` trigger. But they work for all `groups_join_specific_group` triggers as `$args[0]` is the same as `$user_ID`.

**Comment by tw2113** [comment]
I assume you have tested this out with the rest, but I want to just get confirmation there.

Lastly, do you think you could do the pull request against the 19-join-specific-group-fix branch? That fix hasn't been fully merged in yet either, and may as well keep them together.

**Comment by cherbst** [comment]
I have created the new pull request #48. I have tested this fix with group invitation and request memberships, not with all possible triggers. But as this patch mainly adds actions to hooks that were not used before, I don't think it will introduce any regressions.

**Comment by tw2113** [comment]
Closing as we took it in on the other one.

**Pull request closed without merging**